File: 프로그래머스/Level2/기능개발.py
# https://programmers.co.kr/learn/courses/30/lessons/42586

def solution(progresses, speeds):
    days = []
    for i in range(len(progresses)):
        now = 100 - progresses[i]
        day = now // speeds[i]
        if now % speeds[i] != 0:
            day += 1
        days.append(day)
    answer = []
    cnt = 0
    temp = days[0]
    for i in days:
        if temp >= i:
            cnt += 1
        else:
            answer.append(cnt)
            temp = i
            cnt = 1
    if cnt != 0:
        answer.append(cnt)
    return answer

File: 프로그래머스/Level2/기능개발.js
// https://programmers.co.kr/learn/courses/30/lessons/42586

const solution = (p, s) => {
  const days = [];
  p.forEach((tempP, idx) =>
    (100 - tempP) % s[idx] === 0
      ? days.push((100 - tempP) / s[idx])
      : days.push(Math.floor((100 - tempP) / s[idx]) + 1)
  );
  let now = days[0];
  let cnt = 0;
  let answer = [];
  for (const day of days) {
    if (now >= day) cnt += 1;
    else {
      answer = [...answer, cnt];
      cnt = 1;
      now = day;
    }
  }
  return (answer = [...answer, cnt]);
};
